_posts/2021-09-23-reassign-partitions.md

```diff
@@ -0,0 +1,280 @@
+---
+layout: post
+title:  "Reassign partitions in Kafka Cluster"
+date: 2021-09-23
+author: shubham_rawat
+---
+
+Scaling Kafka Brokers, removing old brokers and the increasing no. of topics and partitions are all pretty common tasks for Kafka users.
+All these cases cause the load on the brokers to get unbalanced, so to manage this, the most convenient way is to reassign the partitions among all the brokers.
+In Strimzi, this is easily done with the help of the partition reassignment tool: `kafka-reassign-partitions.sh`.
+
+<!--more-->
+
+## Partition Reassignment Tool
+
+Within a broker pod, the kafka-reassign-partitions.sh tool allows you to reassign partitions to different brokers.
+
+It has three different modes:
+
+| Modes | Description |
+| :-----------: | ------------- |
+| `generate`  | It takes a set of topics and brokers and generates a reassignment JSON file which will result in the partitions of those                     topics being assigned to those brokers. Because this operates on whole topics, it cannot be used when you only want to                       reassign some partitions of some topics.|
+| `execute` | This takes a reassignment JSON file and applies it to the partitions and brokers in the cluster. Brokers that gain partitions                 as a result become followers of the partition leader. For a given partition, once the new broker has caught up and joined                    the ISR (in-sync replicas) the old broker will stop being a follower and will delete its replica. |
+| `verify`  | Using the same reassignment JSON file as the --execute step, --verify checks whether all the partitions in the file have been moved to their intended brokers. If the reassignment is complete, --verify also removes any traffic throttles (--throttle) that are in effect. Unless removed, throttles will continue to affect the cluster even after the reassignment has finished. |
```

Comment:
```suggestion
| `generate`  | Takes a set of topics and brokers and generates a reassignment JSON file which will result in the partitions of those topics being assigned to those brokers. Because this operates on whole topics, it cannot be used when you only want to reassign some partitions for some topics.|
| `execute` | Takes a reassignment JSON file and applies it to the partitions and brokers in the cluster. Brokers that gain partitions as a result become followers of the partition leader. For a given partition, once the new broker has caught up and joined the ISR (in-sync replicas) the old broker will stop being a follower and will delete its replica. |
| `verify`  | Using the same reassignment JSON file as the `--execute` step, `--verify` checks whether all the partitions in the file have been moved to their intended brokers. If the reassignment is complete, `--verify` also removes any traffic throttles (`--throttle`) that are in effect. Unless removed, throttles will continue to affect the cluster even after the reassignment has finished. |
```